### PR TITLE
Implement resize based on randr events

### PR DESCRIPTION
**Pull request by marcusramberg**
Implementing adding and removing monitors is more complicated, but this part seems solid, so I propose adding resizing based on randr events in a initial pass. This PR includes your two initial commits sto setup randr event listening.

### CODE REVIEW

File: src/nimdowpkg/windowmanager.nim [hunk 1]
```diff
@@ -951,6 +959,22 @@ proc onConfigureRequest(this: WindowManager, e: XConfigureRequestEvent) =
 
   discard XSync(this.display, false)
 
+proc onConfigureNotify(this: WindowManager, e: XConfigureEvent) =
+  if e.window == this.rootWindow:
+    log "rootWindow onConfigureNotify"
```
Comment (avahe-kellenberger) [inline src/nimdowpkg/windowmanager.nim:964]: Could we remove the log statement?

**Comment by avahe-kellenberger** [comment]
Thanks for making the PR! What behaviors have you tested as far as randr events go?

File: src/nimdowpkg/windowmanager.nim (hunk 1 above)
Comment (marcusramberg) [reply src/nimdowpkg/windowmanager.nim:964]: I'll fix that.

> Thanks for making the PR! What behaviors have you tested as far as randr events go?

Tested resizing with both xephyr and by moving nimdow between my laptop and an external display while running on a laptop. It won't add/remove displays but it handles the resolution changing.

**Review by avahe-kellenberger — APPROVED** [review]
Looks good, I'm going to merge and do some testing before I do a new release. Thanks for your help!

**Pull request merged**